### PR TITLE
Adds aiohttp handlers for docker registry

### PR DESCRIPTION
**Pull request by dkliban**
The content app can be run in the foreground using the following command:

gunicorn pulpcore.content:server --bind 'localhost:8080' --worker-class 'aiohttp.GunicornWebWorker' -w 2 -t 500 --log-level debug

Then the docker client can pull with the following command:

sudo docker --debug pull localhost:8080/foo

closes #4271
https://pulp.plan.io/issues/4271

### CODE REVIEW

File: pulp_docker/app/registry.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        full_headers['Content-Length'] = os.path.getsize(path)
+        full_headers['Content-Disposition'] = 'attachment; filename={n}'.format(
+            n=os.path.basename(path))
+        file_response = web.FileResponse(path, headers=full_headers)
+        return file_response
+
+    @staticmethod
+    async def serve_v2(request):
+        """
+        Handler for Docker Registry v2 root.
```
[the remark this answers is not in the document]
Comment (ipanova) [reply pulp_docker/app/registry.py]: this endpoint is an equivalent to v1/_ping these are the root endpoits of docker registry api which tells the clients with what api version  he is talking with, tldr, docker client always hits the root endpoint first

[the remark this answers is not in the document]
Comment (asmacdo) [reply pulp_docker/app/registry.py]: I just meant to add this info to the docstring

File: pulp_docker/app/registry.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        tag_name = request.match_info['tag_name']
+        distribution = await Registry.match_distribution(path)
+        accept_headers = await Registry.get_accept_headers(request)
+        if MEDIA_TYPE.MANIFEST_LIST in accept_headers:
+            try:
+                tag = ManifestListTag.objects.get(
+                    pk__in=distribution.publication.repository_version.content,
+                    name=tag_name
+                )
+            # If there is no manifest list tag, try again with manifest tag.
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/registry.py]: You are right. I changed it back.

File: pulp_docker/app/urls.py [hunk 3]
```diff
@@ -1,19 +1 @@
...
-    url(r'^v2/$', VersionView.as_view(), name='registry-root'),
-    url(r'^v2/(?P<path>.+)/tags/list$', TagsListView.as_view()),
-    url(r'^v2/(?P<path>.+)/manifests/(?P<digest>sha256:[A-Fa-f0-9]+)$',
-        BlobManifestView.as_view()),
-    url(r'^v2/(?P<path>.+)/manifests/(?P<tag_name>[A-Za-z0-9\._-]+)$',
-        TagView.as_view()),
-    url(r'^v2/(?P<path>.+)/blobs/(?P<digest>sha256:[A-Fa-f0-9]+)$',
-        BlobManifestView.as_view())
-]
+urlpatterns = []
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/urls.py]: It's not. Will remove.

File: pulp_docker/app/registry.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+        Raises:
+            PathNotResolved: when not matched.
+
+        """
+        try:
+            return DockerDistribution.objects.get(base_path=path)
+        except ObjectDoesNotExist:
+            log.debug(_('DockerDistribution not matched for {path} using: {base_paths}').format(
+                path=path, base_paths=path
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/registry.py]: We don't. That was a copy past error on my part. Will update.

File: pulp_docker/app/registry.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    @staticmethod
+    async def match_distribution(path):
+        """
+        Match a distribution using a base path.
+
+        Args:
+            path (str): The path component of the URL.
+
+        Returns:
+            Distribution: The matched distribution.
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/registry.py]: Thanks!

File: pulp_docker/app/registry.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/registry.py]: Will do

File: pulp_docker/app/registry.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+        Returns:
+            List of media types supported by the client.
+
+        """
+        accept_headers = []
+        for header, value in request.raw_headers:
+            if header == b'Accept':
+                accept_headers.append(value.decode('UTF-8'))
+        return accept_headers
```
[the remark this answers is not in the document]
Comment (dkliban) [reply pulp_docker/app/registry.py]: I'll rename it here and the method name also.

**Comment by ipanova** [comment]
@dkliban @asmacdo i tested and it worked!
- docker pull( it fetches latest if nothing specified)
- docker pull by exiting tag
- docker pull by existing digest 
- docker pull by non existing tag/digest --> 404 not found

**Pull request merged**